string filepath
string datetime
---
Assembly assembly